File: psesca/controller.py
from abc import ABC,abstractmethod
from .factory import Factory

class Controller:
    def __init__(self, climber, stanceGraph):
        self.climber = climber
        self.graph = stanceGraph
        self.simulator = Factory.get().buildSimulator(stanceGraph.getRoute())
        self.simulator.addClimber(climber)

    @abstractmethod
    def tryStep(self, startPosition, startState, endPosition):
        pass

    def tests(self) :
        if any([
            self.simulator.tests()
            ]):
            print("Controller tests failed")
            return True
        else:
            return False

File: psesca/native/controller.cpp
#include "controller.hpp"

Controller::Controller(ClimberModel const * c, StanceGraph const * sg):
	climber(c),
	stanceGraph(sg),
	simulator(new Simulator(sg->getRoute()))
{
	simulator->addClimber(climber);
}

Controller::~Controller()
{
	delete simulator;
}

bool Controller::tests() const
{
	bool r = false;
	r = r || simulator->tests();
	if(r) {
		std::cout << "Controller tests failed." << std::endl;
	}
	return r;
}

